tests/parser-tests.lisp

```diff
@@ -1,21 +1,31 @@
 (in-package #:coalton-tests)
 
 (deftest test-parser ()
-  (let* ((glob (merge-pathnames "tests/parser/*.bad.coalton" (asdf:system-source-directory "coalton/tests")))
+  (labels ((test-files (pattern)
+             (directory (merge-pathnames pattern (asdf:system-source-directory "coalton/tests"))))
 
-         (files (directory glob)))
+           (parse-file (file)
+             (with-open-file (stream file)
+               (parser:with-reader-context stream
+                 (parser:read-program stream (error:make-coalton-file :stream stream :name (namestring file)) :mode :file))))
 
-    (loop :for file :in files
-          :do (with-open-file (stream file)
-                (signals parser:parse-error
-                  (parser:with-reader-context stream
-                    (parser:read-program stream (error:make-coalton-file :stream stream :name (namestring file)) :mode :file))))))
+           (parse-error-text (file)
+             (with-open-file (stream file)
+               (handler-case
+                   (parser:with-reader-context stream
+                     (parser:read-program stream (error:make-coalton-file :stream stream :name "test") :mode :file))
+                 (error:coalton-base-error (c)
+                   (princ-to-string c))))))
+    (loop :for file :in (test-files "tests/parser/*.bad.coalton")
```

Comment:
nit: i prefer either DOLIST as written, or more use of LOOP:

```
(loop :for :in
      :for := 
      :do ...)
```

Reply:
done